src/BaseConnector.ts

```diff
@@ -2,23 +2,23 @@ import { nanoid } from 'nanoid'
 import { Reshuffle } from './types'
 import { EventConfiguration } from './'
 
-class BaseConnector<OptionsType = any> {
+class BaseConnector<ConfigOptionsType = any, EventOptionsType = any> {
```

Comment:
Could this be something like `Record<string,any>` ?

and for `ConfigOptionsType` - should this be `ConnectorOptionsType` ?

Reply:
We could but we'd loose the typing and completion, it will allow anything to be passed as options.
For instance, the HttpConnector will only accept authKey and authScript so you want the implementation to provide a type accordingly.

Reply:
I first prefixed the types with Connector but it's redundant